### PR TITLE
Add unit tests to cover scopes and scope selectors during quota sync logic

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
**What this PR does / why we need it**:
Adds unit tests to cover scopes and scope selectors fields in the quota spec. Existing unit test for quota sync does not cover scopes.

Also while adding tests found a minor issue with 'Exists' scope selector operator. This is also being fixed in this PR.

**Special notes for your reviewer**:

**Release note**:

```release-note
Unit tests for scopes and scope selectors in the quota spec
```
/cc @derekwaynecarr @sjenning @bsalamat @kubernetes/sig-scheduling-pr-reviews @kubernetes/sig-node-pr-reviews

### CODE REVIEW

File: pkg/quota/evaluator/core/pods.go [hunk 1]
```diff
@@ -18,14 +18,13 @@ package core
...
 	"fmt"
-	"strings"
-	"time"
-
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"strings"
```
Comment (k82cn) [inline pkg/quota/evaluator/core/pods.go]: please revert this change.

**Comment by vikaschoudhary16** [comment]
@k82cn done!!

File: pkg/controller/resourcequota/resource_quota_controller_test.go [hunk 2]
```diff
@@ -136,6 +136,66 @@ func newTestPods() []runtime.Object {
...
+		&v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{Name: "pod-running", Namespace: "testing"},
+			Status:     v1.PodStatus{Phase: v1.PodRunning},
+			Spec: v1.PodSpec{
+				Volumes:    []v1.Volume{{Name: "vol"}},
+				Containers: []v1.Container{{Name: "ctr", Image: "image", Resources: getResourceRequirements(getResourceList("", ""), getResourceList("", ""))}},
+			},
+		},
+		&v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{Name: "pod-running-2", Namespace: "testing"},
```
Comment (sjenning) [inline pkg/controller/resourcequota/resource_quota_controller_test.go]: i'm not seeing the difference between this test case and the previous one.

Comment (vikaschoudhary16) [reply pkg/controller/resourcequota/resource_quota_controller_test.go]: @sjenning  Assuming "this test case and the previous one" refering to `newBestEffortTestPods()` and `newTestPods()`, this utility function has two best effort pods while in previous one there is no best effort pods. This utility fucntion is added to test `ResourceQuotaScopeBestEffort`.

Comment (sjenning) [reply pkg/controller/resourcequota/resource_quota_controller_test.go]: yeah, nevermind

**Comment by sjenning** [comment]
LGTM

**Comment by k82cn** [comment]
UT part LGTM :)

File: pkg/quota/evaluator/core/pods.go [hunk 3]
```diff
@@ -380,7 +380,10 @@ func podMatchesSelector(pod *api.Pod, selector api.ScopedResourceSelectorRequire
 	if err != nil {
 		return false, fmt.Errorf("failed to parse and convert selector: %v", err)
 	}
-	m := map[string]string{string(api.ResourceQuotaScopePriorityClass): pod.Spec.PriorityClassName}
+	var m map[string]string
+	if pod.Spec.PriorityClassName != "" {
```
[the remark this answers is not in the document]
Comment (vikaschoudhary16) [reply pkg/quota/evaluator/core/pods.go]: @k82cn in the "quota by priority" logic today there is no special handling for any priority. It does not matter from quota perspective if priority in the pod is "system-cluster-critical" , default or any other. What matters is if priority in quota's scopes matches pod's priority.

[the remark this answers is not in the document]
Comment (k82cn) [reply pkg/quota/evaluator/core/pods.go]: What I means is priority class name maybe updated in other admission controller, how resource quota admission controller handle that?

[the remark this answers is not in the document]
Comment (k82cn) [reply pkg/quota/evaluator/core/pods.go]: LGTM as we slacked; opened https://github.com/kubernetes/kubernetes/pull/66854 to handled "but did not set priority class name to default priority class name" case.

[the remark this answers is not in the document]
Comment (ravisantoshgudimetla) [reply pkg/quota/evaluator/core/pods.go]: I think you could just use a `len(pod.Spec.PriorityClassName) > 0` condition here instead of checking for empty string and that should eliminate the need for having empty string as priorityClass.

[the remark this answers is not in the document]
Comment (vikaschoudhary16) [reply pkg/quota/evaluator/core/pods.go]: @ravisantoshgudimetla Thanks! done!!

[the remark this answers is not in the document]
Comment (ravisantoshgudimetla) [reply pkg/quota/evaluator/core/pods.go]: Thanks @vikaschoudhary16. @k82cn do we still need the PR you introduced with this change? Either way this PR lgtm.

**Review by ravisantoshgudimetla — APPROVED** [review]
Thanks @vikaschoudhary16.

/lgtm
